src/main/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurer.java

```diff
@@ -169,17 +176,28 @@ void updateAdminPassword() {
                 System.exit(-1);
             }
 
-            // Print an update to the logs
-            System.out.println("Admin password set successfully.");
-
+            // Update the custom password in internal_users.yml file
             writePasswordToInternalUsersFile(ADMIN_PASSWORD, INTERNAL_USERS_FILE_PATH);
 
+            System.out.println("Admin password set successfully.");
+
         } catch (IOException e) {
             System.out.println("Exception updating the admin password : " + e.getMessage());
             System.exit(-1);
         }
     }
 
+    /**
+     * Check if the password for admin user was already updated. (Possibly via a custom internal_users.yml)
+     * @param internalUsersFile Path to internal_users.yml file
+     * @return true if password was already updated, false otherwise
+     * @throws IOException if there was an error while reading the file
+     */
+    private boolean isAdminPasswordSetToAdmin(String internalUsersFile) throws IOException {
+        JsonNode internalUsers = YAML_MAPPER.readTree(new FileInputStream(internalUsersFile));
+        return internalUsers.has("admin") && internalUsers.get("admin").get("hash").asText().equals(DEFAULT_ADMIN_PASSWORD_HASH);
```

Comment:
This may be the hash of the [demo configuration admin user](https://github.com/opensearch-project/security/blob/main/config/internal_users.yml#L14), but its still possible to using the [hashing tool](https://github.com/opensearch-project/security/blob/main/tools/hash.sh) to hash `admin` as a password and replace the value. 

If you categorically want to block `admin` as the admin password, you would need to use [OpenBSDBCrypt](https://github.com/opensearch-project/security/blob/d526c9f6c2a438c14db8b413148204510b9fe2e2/src/main/java/org/opensearch/security/auth/internal/InternalAuthenticationBackend.java#L87-L95) to check if the configured password is a hash of `admin`.